mypy/test/data.py

```diff
@@ -149,7 +149,7 @@ def parse_test_case(case: 'DataDrivenTestCase') -> None:
     case.input = input
     case.output = output
     case.output2 = output2
-    case.lastline = item.line
+    case.lastline = item.line+len(item.data)-1
```

Comment:
For readability, please add spaces around the + and - operators.

Also, consider renaming this item to "numlines", and not subtracting 1. That would make the slices in helpers.py cleaner. It would also catch other places where this variable is used or set that might need adjusting.

Reply:
Well, I was not able to understand you completely.
For example-
```
239:[case testSample]
240:line 1
241:line 2
242:line 3
243:[out]
244:error 1
245:error 2
246:error 3
247:
```
**Before my changes**
`case.line=239`<br>
`case.lastline=6`<br>
`test = '\n'.join(data_lines[239:6])`<br>
**After my changes**
`case.line=239`<br>
`case.lastline=6 + 3 (len(item.data)) - 1 = 8 (exact length of the test case)`<br>
`test = '\n'.join(data_lines[ 239 : 239 + ( 6 + 3 - 1 ) - 1 ] ) =  '\n'.join(data_lines[ 239 : 246 ] )`<br>

Now if I change the class property `lastline` to `numlines` but do not subtract 1, then `case.numlines` will be 1 more than the number of lines in the test case and we will have to subtract 2 instead of 1 in the slices in helpers.py.
Instead, I have a suggestion to give `case.lastline` the actual meaning instead of number of lines. This will make the slices in helpers.py as [test.line:test.lastline]. 
For that we'll have to make `case.lastline=case.line + item.line + len(item.data) - 2` 

What do you think would be a better option?

Reply:
Okay, I like that suggestion. I hadn't noticed that both `-1` adjustment pointed in the same direction! The first one presumably compensates for the `[case foobar]` line, which isn't included (`case.line` points to the line after it).

Just rename the variable, please, so we know it's not used in the old sense elsewhere.